### PR TITLE
Fix/nightly ci

### PR DESCRIPTION
**Pull request by smoser**
This is an attempt at fixing nightly c-i similar to https://github.com/project-stacker/stacker/pull/496
but with more re-use and building more nightkly.

### CODE REVIEW

**Comment by hallyn** [comment]
Seems reasonable.

**Comment by smoser** [comment]
So it looks like this doesn't work as intended.

I made the nightly workflow run on PR in the top most commit.
 * [nightly](https://github.com/project-stacker/stacker/actions/runs/6097394094/job/16544941059?pr=497)
 * [c-i](https://github.com/project-stacker/stacker/actions/runs/6097394092/workflow?pr=497)

Both of those show in their test output:
```
ok 19 all container contents must be accounted for # skip test_all_container_contents_must_be_accounted_for is slow. Set SLOW_TEST=true to run.
ok 20 bom tool should work inside run # skip test_bom_tool_should_work_inside_run is slow. Set SLOW_TEST=true to run.
```

Further investigation shows that I had SLOW_TESTS (plural) in the workflow 'env' while the code is looking at SLOW_TEST (singular).

Change made, pushed, lets see.

After this runs i have to still drop the top-most commit so that nightly doesn't run on PR.

File: .github/workflows/nightly.yaml [hunk 1]
```diff
@@ -4,31 +4,10 @@ on:
...
-          podman run -d -p ${ZOT_PORT}:${ZOT_PORT} ghcr.io/project-zot/zot-minimal-linux-amd64:latest
-          # check if reachable
-          while true; do x=0; curl -f http://${REGISTRY_URL}/v2/ || x=1; if [ $x -eq 0 ]; then break; fi; sleep 1; done
-      - name: test
-        env:
-          SLOW_TEST: true
-        run: |
-          make check
+  build:
+    uses: ./.github/workflows/build.yaml
```
Comment (rchincha) [inline .github/workflows/nightly.yaml:13]: where is "slow-test" actually set to true?

Comment (smoser) [reply .github/workflows/nightly.yaml:13]: > where is "slow-test" actually set to true?

It defaults to true. (.github/workflows/build.yaml).

it seems that "run all the tests" is a better default.  That way in a release and in a nightly we get all the tests run. the only place we don't get them all run is in the 'ci.yaml' caller.

**Comment by smoser** [comment]
both of the slow test runs failed. see [here](https://github.com/project-stacker/stacker/actions/runs/6100746015/job/16555687757?pr=497).

unpriv failed:
```

**Comment by smoser** [comment]
I'd like to ask that this be landed.  nightly c-i will still be broken, but this PR gets us closer.
Once landed, I think you'll be able to run the nightly manually on a branch to debug/fix it.
See https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow for how to do that.

**Pull request merged**